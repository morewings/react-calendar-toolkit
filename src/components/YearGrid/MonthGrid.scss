@import "./../../variables";

.month-grid-wrapper {
  display: flex;
  width: $innerWidth;
  flex-direction: row;
  flex-wrap: wrap;
  padding: 0 $innerPadding $innerPadding;
}

.month {
  text-align: center;
  box-sizing: border-box;
  flex: $monthCellFlex;
  height: $dayHeight;
  line-height: $dayHeight;
  border: 1px solid transparent;
  cursor: pointer;
  border-radius: $borderRadius;
  &:active{
    outline: none;
  }
  &:hover{
    background-color: $lightGray;
  }
  &.disabled {
    color: $disabled;
  }
  &.isSelected {
    color: white;
    background-color: $active;
  }
  &.isToday {
    border-color: $gray;
  }
}