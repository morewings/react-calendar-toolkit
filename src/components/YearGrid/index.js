export {default} from './YearGrid';
export {default as useScrollIntoView} from './useScrollIntoView';
