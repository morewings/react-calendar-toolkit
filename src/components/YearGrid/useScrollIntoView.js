import {useEffect} from 'react';

const defaultOptions = {
  block: 'center',
  inline: 'center',
  behavior: 'smooth',
};

const useScrollIntoView = (ref, isSelected, options = defaultOptions) => {
  useEffect(() => {
    if (isSelected) {
      ref.current.scrollIntoView(options);
    }
  }, [ref, isSelected, options]);
};

export default useScrollIntoView;
