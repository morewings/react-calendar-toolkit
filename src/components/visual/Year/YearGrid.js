import React from 'react';
import PropTypes from 'prop-types';
import classes from './Year.module.css';

const YearGrid = props => (
  <div className={classes.scrollContainer}>
    <div className={props.className}>{props.children}</div>
  </div>
);

YearGrid.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

YearGrid.defaultProps = {
  className: classes.wrapper,
};

export default YearGrid;
