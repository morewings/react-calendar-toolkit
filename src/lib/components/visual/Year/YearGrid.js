import React from 'react';
import PropTypes from 'prop-types';
import classes from './Year.module.css';

const YearGrid = ({children, className}) => (
  <div className={classes.scrollContainer}>
    <div className={className || classes.grid}>{children}</div>
  </div>
);

YearGrid.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
};

export default YearGrid;
