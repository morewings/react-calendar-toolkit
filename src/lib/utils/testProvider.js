import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'lib/features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'lib/features/theme/withTheme';
import {Provider as ModalProvider} from 'lib/features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'lib/features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
