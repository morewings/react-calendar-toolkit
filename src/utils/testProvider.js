import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'features/theme/withTheme';
import {Provider as ModalProvider} from 'features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
