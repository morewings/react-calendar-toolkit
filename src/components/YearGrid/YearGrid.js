import React, {useCallback} from 'react';
import PropTypes from 'prop-types';
import {useDispatch, useSelector} from 'react-redux';
import {actionTypes, selectors} from 'features/datepicker';
import {getTime, getYears, getIsSameYear} from 'utils/dateUtils';
import Year, {YearWrapper} from 'components/visual/Year';

const YearGrid = props => {
  const dispatch = useDispatch();
  const selectedTimestamp = useSelector(selectors.getSelectedTimestamp);
  const todayTimestamp = useSelector(selectors.getTodayTimestamp);
  const years = getYears();
  const setYear = useCallback(
    date =>
      dispatch({
        type: actionTypes.SET_DATE,
        payload: {
          selectedTimestamp: getTime(date),
          precision: 'month',
        },
      }),
    [dispatch]
  );
  const Wrapper = props.wrapperElement;
  return (
    <Wrapper className={props.wrapperClassname}>
      {years.map(({yearNumber, date}) => (
        <Year
          disabled={false}
          date={date}
          onSetYear={setYear}
          isSameYear={getIsSameYear(date, todayTimestamp)}
          isSelected={getIsSameYear(date, selectedTimestamp)}
          key={yearNumber}
          yearNumber={yearNumber}
        />
      ))}
    </Wrapper>
  );
};

YearGrid.propTypes = {
  wrapperClassname: PropTypes.string,
  wrapperElement: PropTypes.elementType,
};

YearGrid.defaultProps = {
  wrapperClassname: 'year-grid-wrapper',
  wrapperElement: YearWrapper,
};

export default YearGrid;
