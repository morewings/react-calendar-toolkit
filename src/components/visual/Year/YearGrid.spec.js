import React from 'react';
import {DatepickerMockProvider as Provider} from 'utils/testProvider';
import {render} from '@testing-library/react';
import YearGrid from './YearGrid';

const Mock = () => <div>Mock</div>;

describe('MonthGrid', () => {
  it('renders', () => {
    const {asFragment} = render(
      <YearGrid>
        <Mock />
      </YearGrid>,
      {
        wrapper: ({children}) => <Provider>{children}</Provider>,
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
