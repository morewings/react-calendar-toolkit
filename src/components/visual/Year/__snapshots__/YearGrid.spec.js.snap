// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthGrid renders 1`] = `
<DocumentFragment>
  <div
    class="scrollContainer"
  >
    <div
      class="grid"
    >
      <div>
        Mock
      </div>
    </div>
  </div>
</DocumentFragment>
`;
